assets/helpers/paymentIntegrations/readerRevenueApis.js

```diff
@@ -145,6 +149,9 @@ function checkRegularStatus(participations: Participations, csrf: CsrfState): Ob
   };
 
   return (json) => {
+    if (json.guestAccountCreationToken) {
+      dispatch(setGuestAccountCreationToken(json.guestAccountCreationToken));
```

Comment:
Having `dispatch` here is an abstraction leak that makes this module less portable. Let's realise that all we need
here is a callback that accepts `json.guestAccountCreationToken`, so a function `string => void`. I would define
it as such here, and then at the calling site you can write:
```
postRegularStripeRequest(
  data, 
  particiaptions, 
  csrf, 
  (token) => dispatch(setGuestAccountCreationToken(token))
)
```